### PR TITLE
Add Image Upload to Profile Creation/Edit and Recipe Posting

### PR DESCRIPTION
**Pull request by bessmarche**
The **Blobstore API** works by adding an extra step in a request. Normally, the action of the form is a URL that maps to a servlet on the server. When the user submits the form, the servlet can get the values entered by the user through the request.getParameter() function, and can then store those values.
But when you use Blobstore, the action of the form is a URL that points to Blobstore. When the user submits the form, the request first goes to Blobstore. Blobstore takes care of handling the file uploaded by the user, and then forwards the request to your servlet. Your servlet can then get the data entered by the user, including the URL of the file that Blobstore handled.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/ProfileEditServlet.java [hunk 1]
```diff
@@ -14,16 +14,29 @@
 
 package com.google.sps.servlets;
```
Comment (donpaul999) [inline src/main/java/com/google/sps/servlets/ProfileEditServlet.java]: We can't have the same file for Profile Creation & Edit? I can see the code is very similar.

File: src/main/webapp/profile_edit.jsp [hunk 2]
```diff
@@ -1,13 +1,22 @@
 <%@page contentType="text/html" pageEncoding="UTF-8"%>
```
Comment (donpaul999) [inline src/main/webapp/profile_edit.jsp]: The same comment as the before. Maybe there is a way to remove duplicate code and to create a template for both pages.

File: src/main/webapp/recipe_post.jsp [hunk 3]
```diff
@@ -1,14 +1,25 @@
 <%@page contentType="text/html" pageEncoding="UTF-8"%>
 <%@taglib prefix="t" tagdir="/WEB-INF/tags" %>
 <%@taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+<%@ page import="com.google.appengine.api.blobstore.BlobstoreServiceFactory" %>
+<%@ page import="com.google.appengine.api.blobstore.BlobstoreService" %>
+<%@ page import="com.google.appengine.api.blobstore.BlobstoreServiceFactory" %>
+<% BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
```
Comment (donpaul999) [inline src/main/webapp/recipe_post.jsp:7]: I see that you are including this in 3 different pages. You should have a different file for this and only call it from every page.

**Review by donpaul999 — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/ProfileEditServlet.java (hunk 1 above)
Comment (bessmarche) [reply src/main/java/com/google/sps/servlets/ProfileEditServlet.java]: I'll sort this out in a separate Pull Request

File: src/main/webapp/profile_edit.jsp (hunk 2 above)
Comment (bessmarche) [reply src/main/webapp/profile_edit.jsp]: Same as above

**Review by ramaroberto — APPROVED** [review]

**Pull request merged**